### DISCUSSION TITLE
I updated brew and got the following after restarting Emacs

### DISCUSSION THREAD

**Discussion by NatHarari** [minad/jinx#118] [Q&A] [answered]
As per the title: I updated brew and got the following after restarting Emacs:

```
gcc -I. -O2 -Wall -Wextra -fPIC -shared -o jinx-mod.dylib jinx-mod.c -I/usr/include/enchant-2 -lenchant-2
jinx-mod.c:19:10: fatal error: 'enchant.h' file not found
#include <enchant.h>
         ^~~~~~~~~~~
1 error generated.
Jinx: Compilation of jinx-mod.dylib failed
```

Enchant version: @(#) International Ispell Version 3.1.20 (but really Enchant 2.6.2)

**Comment by NatHarari** [comment]
I forgot to add:
  jinx                           20231111.850   installed             Enchanted Spell Checker

**Comment by minad** [comment]
I am not a Mac/Brew user, but I think you are missing pkgconf or pkg-config, such that Enchant is not detected properly.

**Reply by NatHarari** [reply]
In fact, it was something I checked for before posting here. I do have it installed. But yes, it might have been related which is why I searched as well before posting here. 😀

**Comment by NatHarari** [comment]
I finally had time to go back and check it out. Here's the problem printed from when I tried it in terminal instead of when Emacs tries to do it:

```
gcc -I. -O2 -Wall -Wextra -fPIC -shared -o jinx-mod.dylib jinx-mod.c -I/usr/include/enchant-2 -lenchant-2 
clang: error: no such file or directory: 'jinx-mod.c'
```

Now, you'll notice it's trying to find `usr/include/enchant-2` but this is an M chip so the location is different: `/opt/homebrew/bin` and packages typically are also installed in `/opt/homebrew/Cellar` so, in this case:
`/opt/homebrew/Cellar/enchant/2.6.2/bin/enchant-2`

I'm hoping this all makes sense. I'm not certain what else I should do. Please let me know if there is anything else I can give to you for information.

Also, I reinstalled png-config and it did not help. Just letting you know. I'm thinking the path is incorrect in the Emacs package somehow for where to search, but I'm unsure.

**Comment by NatHarari** [comment, marked as answer]
I got it to work. Here was the issue and here is the solution:
Jinx relies on Enchant, and Enchant was updated. Brew installs directories with a hierarchy which includes the latest version number, which throws off the installer. Here was the solution to the problem:

`gcc -I. -I/opt/homebrew/Cellar/enchant/2.6.2/include/enchant-2 -O2 -Wall -Wextra -fPIC -shared -o jinx-mod.dylib jinx-mod.c /opt/homebrew/Cellar/enchant/2.6.2/lib/libenchant-2.dylib`

You will also want your PATHS correctly done in your .zshrc or your .bashrc. Here are mine:

```
export LIBRARY_PATH="/opt/homebrew/opt/enchant/lib/:$LIBRARY_PATH"
export C_INCLUDE_PATH="/opt/homebrew/Cellar/enchant/2.6.2/include/enchant-2:$C_INCLUDE_PATH"
```

After you have added that to your .zshrc/.bashrc/whateverfishuses/anyotherkindofshell, you will want to:

`source ~/.zshrc`
(or equivalent for your file)

This will reload the paths. Do that before running the `gcc` line.

As you can see, the version number is included in the directory structure. So, for anyone who is searching why Jinx isn't working after updating brew, the solution is to find the equivalent of those directories (the number 2.6.2 will be different), and run that line. It will work. The full structure is there to facilitate those looking for a solution whom, like me, are not adept at code.
Note: If you are on an Intel, the path which Brew installs to is different than on an M chip like I have, so you will have to look up how brew installs stuff on your Intel.

I hope this helps others in the future.
